sstables/sstable_set.cc

```diff
@@ -177,6 +177,26 @@ sstable_set::insert(shared_sstable sst) {
     _impl->insert(sst);
 }
 
+// Default implementation for sstable_set_impl vector insert
+void
+sstable_set_impl::insert(const std::vector<shared_sstable>& ssts) {
+    try {
+        for (const auto& sst : ssts) {
+            insert(sst);
+        }
+    } catch (...) {
+        for (const auto& sst : ssts) {
+            erase(sst);
+        }
```

Comment:
It's more correct to erase only the ones that were successfully inserted.

Reply:
Agreed.  We can maintain an undo vector for each successfully inserted sstable and erase only those.

Reply:
I also thought about this during my review, but I didn't comment on it because pre-existing single-sstable `insert` also can erase an sstable that it didn't insert.

This implementation is OK as long as its forbidden to attempt an insert of an sstable that was already inserted, which I guess is true. But in general it's weird for something called a "set" to break down when the same element is inserted twice.

Reply:
We never insert the same element twice (that I know of). We should consider erroring out in that case, because something went wrong, and to rename the container to satisfy the pedants.